pkg/ingester/wal.go

```diff
@@ -563,12 +585,15 @@ func processCheckpoint(name string, userStates *userStates, params walRecoveryPa
 Loop:
 	for reader.Next() {
 		s := seriesPool.Get().(*Series)
-		if err := proto.Unmarshal(reader.Record(), s); err != nil {
+		m, err := decodeRecord(reader.Record(), CheckpointRecordType1, s)
```

Comment:
So what would happen if the checkpoint is w/o a record? From what I can see it returns an error.